Mage/src/main/java/mage/game/GameState.java

```diff
@@ -42,6 +42,8 @@
 import org.apache.log4j.Logger;
 
 import java.io.Serializable;
+import java.lang.reflect.Constructor;
```

Comment:
Do not use reflect code in game engine (outside of tests and plugins/class loaders). Use interfaces and basic classes instead (GameEvent classes allow to do it).

Reply:
Well you have to explain how to do that.
I can't make an abstract static method in an interface and/or abstract class that would get Overriden by the inheritors.

Reply:
I agree that it is best to avoid reflection if possible, though I will note that `Watcher` class (core game engine) does use reflection already.

(Haven't yet thought about how to make this work with parameterized types without reflection.)